npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser/lockfile_parser.rb

```diff
@@ -151,11 +151,12 @@ def recursively_fetch_npm_lock_dependencies(object_with_dependencies)
 
           object_with_dependencies.
             fetch("dependencies", {}).each do |name, details|
-              next unless semver_version_for(details["version"])
+              version = semver_version_for(details["version"])
```

Comment:
👍

Reply:
I noticed a few places where we could memoize work in the npm code. Since npm seems to be one of our slower ecosystems, it felt like a safe and beneficial change to make. Although, I recognize that this change isn't necessary for resolving this particular defect.